File: backend/src/main/java/com/ryadovoy/todo/task/Task.java
package com.ryadovoy.todo.task;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
public class Task {
    @Id
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "task_seq"
    )
    @SequenceGenerator(
            name = "task_seq",
            allocationSize = 1
    )
    private Long id;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String description;

    @Column(nullable = false)
    private Boolean completed = false;

    public Task(String title, String description) {
        this.title = title;
        this.description = description;
    }
}

File: android/app/src/main/java/com/ryadovoy/todo/data/Task.kt
package com.ryadovoy.todo.data

data class Task(
    val id: Long,
    val title: String,
    val description: String,
    val completed: Boolean
)

data class TaskCreationRequest(
    val title: String,
    val description: String
)

data class TaskUpdateRequest(
    val title: String,
    val description: String,
    val completed: Boolean
)
